### PR TITLE
Read file directly from location

### LINKED ISSUE #6263: On desktop, drag and drop file on the scene should read it directly from its location.
### Discord username

_No response_

### How important is this feature to you?

2 – Somewhat important, but I can live without it

### Describe the idea you'd like to see implemented.

Currently, while dragging and dropping the file on the scene the file is automatically copied to the `data` folder inside a project folder. 

Instead of this on desktop we should:
1. link the file by default and read it,
2. when dragging and dropping with `ctrl` the file should be copied to a `data` folder in a project.

### Is your feature request related to a problem?

_No response_

### Screenshots, screencasts, mockups.

_No response_

### Would you be willing to help us implement this feature?

No

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Closes #6263

### Important Notes
- Still needs to be tested on Windows and macOS
- Currently does not check whether <kbd>Ctrl</kbd> is pressed to forcibly upload the file

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/GraphEditor/upload.ts [hunk 1]
```diff
@@ -66,20 +72,24 @@ export class Uploader {
...
+      isOnLocalBackend,
       stackItem,
     )
     return instance
   }
 
-  async upload(): Promise<string> {
+  async upload(): Promise<UploadResult> {
+    // This non-standard property is defined in Electron.
+    if ('path' in this.file && typeof this.file.path === 'string') {
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/upload.ts]: Should there also be a check for `this.isOnLocalBackend`? Uploading through electron should still work for remote connections after all (e.g. cloud).

**Review by Frizi — APPROVED** [review]
Looks good, apart from one condition that looks to be missing.

File: app/gui2/src/components/GraphEditor/upload.ts (hunk 1 above)
Comment (somebody1234) [reply app/gui2/src/components/GraphEditor/upload.ts]: yes, indeed. <kbd>Ctrl</kbd> key handling is also missing, i've just added that in.
i will be testing on Windows and with ctrl key, but ideally it should be tested on macos too

**Comment by somebody1234** [comment]
ah... forgot to escape `\` -> `\\` on windows...

**Comment by somebody1234** [comment]
linux confirmed to work with and without ctrl. unable to test on cloud, i'm not sure if GUI2 actually works on cloud yet. however, given that it still works locally, i will assume that `localhost` detection is working fine.

**Comment by somebody1234** [comment]
windows also confirmed to work now, with and without ctrl!!!

**Review by vitvakatu — APPROVED** [review]
Works as intended on MacOS

**Pull request merged**